llvm/test/tools/llvm-debuginfo-analyzer/WebAssembly/README.txt

```diff
@@ -0,0 +1,54 @@
+LLVM WebAssembly limitations
+----------------------------
+LLVM does not support any conversion between the binary format (.wasm)
+and the text format (.wat):
```

Comment:
True, but llvm does support its own `.s` format which is very similar to wat.  Would that be useful here instead of wat?

Reply:
You are correct in relation to the `.s` format.

All the references to `.wat` are just for illustrative purposes and the included `.wat` files are not required or used by any of the tests. I am happy to exclude them.

The llvm-debuginfo-analyzer only supports `.wasm`.

Reply:
A concern that I communicated to Carlos offline is that we should not be committing binaries with no clear way to reproduce them from source. Without the .wat files the .wasm files are opaque binaries. Now, if it's sufficient to provide .c/.cpp source and use Clang to regenerate the .wasm, I'm okay with that, but I don't ever work with WebAssembly so I don't know how we usually do this.
@dwblaikie tends to have opinions about this sort of thing, I'll defer to him.

Reply:
I'm fine with including a way to rebuild any wasm binaries, but why not use the `.s` format, which we use elsewhere in llvm?

We don't currently have any usage or reference to wat in llvm and I think it would consistent to keep it that way.

When writing tests for the wasm binary format we tent to use wither `.yaml` files that convert to wasm objects using `objywaml` or `.s` files that convert to wasm using `llvm-mc`.    Can either of these two formats for for you? 

If you are are starting from `.c` sources that you should be able to simplest way to generate the `.s` is simply to run clang with `-S`.

Reply:
> A concern that I communicated to Carlos offline is that we should not be committing binaries with no clear way to reproduce them from source. Without the .wat files the .wasm files are opaque binaries. Now, if it's sufficient to provide .c/.cpp source and use Clang to regenerate the .wasm, I'm okay with that, but I don't ever work with WebAssembly so I don't know how we usually do this. @dwblaikie tends to have opinions about this sort of thing, I'll defer to him.

@pogo59 Thanks for your valuable input.

Reply:
@sbc100 Thanks for your suggestions to use `obj2yaml` and `llvm-mc`.
As I have already the object files, I took the option to use `obj2yaml`.

Uploaded patch:
- Removed the .wasm and .wat files.
- Removed the README.txt as it is irrelevant now.
- Added .yaml files for the tests.